### ISSUE TITLE
Bat autocomplete with zsh is buggy

### ISSUE THREAD

**Issue by alichtman** [sharkdp/bat#495]
Whenever I enter `$ bat <TAB>` to prompt zsh autocompletion, it autocompletes `cache` instead of cycling through available files in the directory, like `zsh` does for other programs.

Typing `$ bat <First Letters>` results in zsh saying no completions can be found.

I've confirmed that cat does not exhibit this behavior, and none of the other tools I've installed with brew do this either.

Here's the result of `$ cat <TAB> <TAB> <TAB>`.

<img width="445" alt="image" src="https://user-images.githubusercontent.com/20600565/52210935-018ba580-284e-11e9-9d47-3d92ea8d9042.png">

Here's `$ bat <TAB> <TAB> <TAB>`

<img width="949" alt="image" src="https://user-images.githubusercontent.com/20600565/52210979-17996600-284e-11e9-8b8d-e7f9327d766a.png">

The issue here is that `bat` prefers tabbing into the options over tabbing into the files it can print out.

OS: macOS Mojave 10.14.1
Installed through HomeBrew
Shell: zsh 5.7 (x86_64-apple-darwin18.2.0)

```
➜ brew info bat                                                                                                                               📦 v1.1.7 with ⬢ v11.9.0 at 7:15AM 
bat: stable 0.9.0 (bottled)
Clone of cat(1) with syntax highlighting and Git integration
https://github.com/sharkdp/bat
/usr/local/Cellar/bat/0.9.0_1 (9 files, 5.5MB) *
  Poured from bottle on 2019-02-03 at 09:38:19
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/bat.rb
==> Dependencies
Build: rust ✘
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d

zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
==> Analytics
install: 11,582 (30 days), 26,041 (90 days), 74,443 (365 days)
install_on_request: 11,553 (30 days), 25,986 (90 days), 74,335 (365 days)
build_error: 0 (30 days)
```

**Comment by alichtman** [comment]
I upgraded to zsh 5.7.2 from 5.7.1 and can't replicate the `$ bat <TAB> <TAB> <TAB>` example from above. So maybe it's a zsh issue and not a bat issue?

I've just run into another issue which stops me from `bat`-ting multiple files at a time with autocomplete. 

`$ bat <Tab to autocomplete first file> <Tab>`

The second tab always fills in `cache` for me.

Another issue I discovered is that when you autocomplete `cache`, the next thing autocompleted is `cache`, a second time. Only after that second cache will zsh give me the option to scroll through options.

<img width="949" alt="image" src="https://user-images.githubusercontent.com/20600565/52212441-1c601900-2852-11e9-9e7c-8bc545070265.png">

Upon selection, the command is invalid.

<img width="739" alt="image" src="https://user-images.githubusercontent.com/20600565/52212465-27b34480-2852-11e9-927c-99287fd42ee9.png">
